jetbrains-core/src/software/aws/toolkits/jetbrains/services/telemetry/TelemetryService.kt

```diff
@@ -119,26 +71,36 @@ class DefaultTelemetryService(settings: AwsSettings) :
         batcher.shutdown()
     }
 
-    override fun record(
-        metricEventMetadata: TelemetryService.MetricEventMetadata,
-        buildEvent: MetricEvent.Builder.() -> Unit
-    ): MetricEvent {
+    fun record(metricEventMetadata: MetricEventMetadata, buildEvent: MetricEvent.Builder.() -> Unit) {
         val builder = DefaultMetricEvent.builder()
-        buildEvent(builder)
         builder.awsAccount(metricEventMetadata.awsAccount)
         builder.awsRegion(metricEventMetadata.awsRegion)
-        val event = builder.build()
-        batcher.enqueue(event)
-        return event
+
+        buildEvent(builder)
+
+        batcher.enqueue(builder.build())
     }
 
-    override suspend fun sendFeedback(sentiment: Sentiment, comment: String) {
+    suspend fun sendFeedback(sentiment: Sentiment, comment: String) {
         publisher.sendFeedback(sentiment, comment)
     }
 
-    private fun record(event: MetricEvent.Builder.() -> Unit): MetricEvent = record(TelemetryService.MetricEventMetadata(), event)
+    private fun record(event: MetricEvent.Builder.() -> Unit) = record(MetricEventMetadata(), event)
 
-    private fun record(metricName: String): MetricEvent = record(TelemetryService.MetricEventMetadata()) {
+    private fun record(metricName: String) = record(MetricEventMetadata()) {
         this.datum(metricName)
     }
+
+    companion object {
+        private const val TELEMETRY_KEY = "aws.toolkits.enableTelemetry"
+        private val TELEMETRY_ENABLED = System.getProperty(TELEMETRY_KEY)?.toBoolean() ?: true
```

Comment:
should be opt-in? or is that later?

Reply:
This is just our backdoor for the sandbox, its not really meant to be used by users IMO